lib/services/suite/suite.go

```diff
@@ -67,6 +67,7 @@ type TestCAConfig struct {
 
 // NewTestCAWithConfig generates a new certificate authority with the specified
 // configuration
+// Keep this function in-sync with lib/auth/auth.go:newKeySet().
```

Comment:
What about pulling `newKeySet()` into lib/services and calling it here? You could abstract its use of `keyStore` with an interface and mock it

Reply:
I don't see the TODO. Rather than refactor the whole existing interface, you could just define a smaller one for `newKeySet` which only includes those that are actually necessary (`NewSSHKeyPair`, `NewTLSKeyPair`, and `NewJWTKeyPair`.) You don't need to mock all of `keystore.KeyStore`. For all that Go's type system lacks, it's nice that interfaces are automatically satisfied for cases like this.

When creating a test method, and then testing that the test method and the real method are in sync, it's just worth suggesting an alternative. I'm okay with this though if you want to merge it, I'll approve, just re-request review please if you change this

Reply:
> I don't see the TODO.

My IDE did this to me for a second time today. Push didn't push 😒 Fixed

>Rather than refactor the whole existing interface, you could just define a smaller one for newKeySet which only includes those that are actually necessary (NewSSHKeyPair, NewTLSKeyPair, and NewJWTKeyPair.) You don't need to mock all of keystore.KeyStore. For all that Go's type system lacks, it's nice that interfaces are automatically satisfied for cases like this.

That was kind of my idea. Extract a smaller interface that satisfies the `newKeySet()` functionality and embed it back to the `keystore.KeyStore`. Basically, the same way as Go's `io` package does.